strawberryfields/engine.py

```diff
@@ -494,14 +485,16 @@ class RemoteEngine:
 
     POLLING_INTERVAL_SECONDS = 1
     VALID_TARGETS = ("X8_01", "X12_01", "X12_02")
+    DEFAULT_TARGETS = {"X8": "X8_01", "X12": "X12_01"}
 
-    def __init__(self, target: str, connection: Connection = Connection()):
+    def __init__(self, target: str, connection: Connection = Connection(), backend_options: dict = None):
```

Comment:
Could be nice to reflect this in the docstring, also, adding the fact that this overrides any `run_options`, should there be options in both (if I understand correctly the change in `io.p`)

Reply:
Added it to the docstring!

The distinction between backend options and run options actually comes from the LocalEngine:

* `backend_options`: global options for the entire backend, required when initializing a backend (e.g., `cutoff_dim`)

* `run_options`: options specified during the backend execution; these are passed directly to the operation `apply` methods (`shots`, etc.).

The blackbird program however doesn't distinguish; backend options and run options are merged into a single dictionary, hence why it looks like they are being overwritten.

For the remote engine, distinguishing between backend options and run options is more semantics than the local engine, since both options apply remotely during `run`. We might revisit this in an ADR. For now, though, since this is a bugfix, I think it's enough to align remote engine with the local engine.